### PR TITLE
Add terraform needed for MessageReceiver, terraform documentation and scripts

### PR DESCRIPTION
**Pull request by bemwen**
## Description

The purpose of this PR is to add the terraform needed to setup the infrastructure needed by the MessageReceiver Azure Function.

This PR:
* Adds Azure Function to terraform deployment
* Adds module needed for the ping tests
* Adds documentation on how to run terraform locally against your sandbox
* Adds script from charges for deploying from local machine to your sandbox
* Adds temporary terraform files to .gitignore for the infrastructure folder
* Change infrastructure project name to "tseries" to shorten it.

The GitHub workflows will come in a separate PR.

## References

* #37

### CODE REVIEW

**Review by lasrinnil — APPROVED** [review]

File: build/infrastructure/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## Variables in `localhost.tfvars`
+
+This describes what values to use for the various values in the `localhost.tfvars` file.
+
+It is assumed that `Azure CLI` for Windows and the Azure AD module for `powershell` is installed. If not, guides can be found found later in this file.
+
+### `environment`
+
+The value of this need to be 1-3 characters.
```
Comment (HenrikSommer) [inline build/infrastructure/README.md:44]: Use of _deploy-from-localhost.cmd_ enforces environment "s"

**Review by HenrikSommer — APPROVED** [review]

File: build/infrastructure/README.md (hunk 1 above)
Comment (BjarkeMeier) [reply build/infrastructure/README.md:44]: Why is that? You're free to set it to anything you please in `localhost.tfvars`.

Comment (bemwen) [reply build/infrastructure/README.md:44]: Alot of the resources has a max length of 24 characters. Between all our variables used in the naming, it means that some should be limited in length and scope to make sure we do not go over the 24 character boundry when the variable is actually used. 

It was calculated earlier going backwards. 24 characters. 4 are used for organization, 7 are reserved for project, 3 reserved for environment, leaving 10 for actual naming.

Comment (BjarkeMeier) [reply build/infrastructure/README.md:44]: My comment was for Henrik. Sorry for not being clear about that.

Comment (bemwen) [reply build/infrastructure/README.md:44]: No worries, answered anyways ;)

Comment (bemwen) [reply build/infrastructure/README.md:44]: Anyway, the answer to that question, is that our script for deploying code into the functions on their sandbox assumes a value of "s" right now. It is not a variable in the script. As such, I added a comment stating that fact to the document so ease the work for people that want to use that script.

Comment (BjarkeMeier) [reply build/infrastructure/README.md:44]: Ahh, the other script with the same name :-D Sorry - I was only thinking of the deployment script for terraform.

**Pull request merged**